tools/go_marshal/marshal/marshal_impl_util.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marshal
+
+import "gvisor.dev/gvisor/pkg/usermem"
+
+// DummyMarshallable implements the Marshallable interface.
+// DummyMarshallable relies on user of this struct to actually
+// implement the functions. It will panic when unimplemented
+// function is called.
+type DummyMarshallable struct{}
```

Comment:
nit: Can we call it `StubMarshallable` instead? Dummy somewhat implies this implements a fake version of marshallable.